tensorflow/python/keras/applications/efficientnet_weight_update_util.py

```diff
@@ -252,56 +257,57 @@ def keras_name_to_tf_name_block(keras_name, keras_block='block1a', tf_block='blo
         tf_name.append(x)
   if use_ema:
     tf_name.append('ExponentialMovingAverage')
-  return '/'.join(tf_name) + ':0'
+  return '/'.join(tf_name)
 
 
-def check_match(keras_block, tf_block, keras_weights, tf_weights):
+def check_match(keras_block, tf_block, keras_weight_names, tf_weight_names, model_name_tf):
   """ Check if the weights in h5 and ckpt match
   
-  we match each name from keras_weights that is in keras_block 
-  and check if there is 1-1 correspondence to names from tf_weights
+  we match each name from keras_weight_names that is in keras_block 
+  and check if there is 1-1 correspondence to names from tf_weight_names
   that is in tf_block
   
   Args:
     keras_block: str, the block name for keras implementation (e.g. 'block1a')
     tf_block: str, the block name for tf implementation (e.g. 'blocks_0')
-    keras_weights: list of str, each string is a name for weights in keras implementation
-    tf_weights: list of str, each string is a name for weights in tf implementation
+    keras_weight_names: list of str, each string is a name for weights in keras implementation
+    tf_weight_names: list of str, each string is a name for weights in tf implementation
   """
```

Comment:
I think rather than adding and removing from the list, can we populate two lists, one from keras weights and one from tf weights, and compare diff? That will allow us to raise an error when all mismatch items, rather than just one.

Reply:
Thanks for the suggestion. Change is made.